ext/mbstring/tests/gh10648.phpt

```diff
@@ -0,0 +1,29 @@
+--TEST--
+GH-10648 (mb_check_encoding() returns true for incorrect but interpretable ISO-2022-JP byte sequences)
+--EXTENSIONS--
+mbstring
+--FILE--
+<?php
+
+$jis_bytes = '1b244224221b2842'; // 'あ' in ISO-2022-JP
+$jis_bytes_without_esc = '1b24422422'; // 'あ' in ISO-2022-JP without trailing escape sequence
+
+$esc_kana = '1b2849211b2842'; // 'ｱ' in ISO-2022-JP (JIS X 0201 7bit kana with escape sequence)
+$so_kana = '0e210f'; // 'ｱ' in ISO-2022-JP (JIS X 0201 7bit kana with SO/SI)
+$gr_kana = 'b1'; // 'ｱ' in ISO-2022-JP (JIS X 0201 8bit kana)
+
+// Testing for trailing escape sequence
+var_dump(mb_check_encoding(hex2bin($jis_bytes), 'JIS'));
+var_dump(mb_check_encoding(hex2bin($jis_bytes_without_esc), 'JIS')); // false
+
```

Comment:
Included of last SI(`0x0f`) to `var_dump(mb_check_encoding(hex2bin("1b244224220f"), "JIS"));` result is true. Originally, I think require escape sequence (`0x1b2842`).

[3v4l](https://3v4l.org/sRnaE) result is PHP 8.0 is returns false. Hmm...

Reply:
Thanks for your review!
Certainly, escape sequences and SO/SI should not be confused.
However, I do not know if this can even be called an irregularity as ISO-2022-JP.
Because RFC 1468 only requires ending with ASCII.
Well, it is natural since SO/SI is not defined in RFC 1468 😢

Reply:
Fixed mb_check_encoding to return false when SO/SI and escape sequences are mismatched.
This fix also causes mb_check_encoding to return false when 0x0e or 0x0f appears by itself, but I think it did not a problem since it is returning false in PHP 8.0.
https://3v4l.org/DhZde